holoscribe/src/scriber/mod.rs

```diff
@@ -17,69 +19,112 @@ impl DebugScriber {
 impl HoloPointStrategy for DebugScriber {
-    fn scribe_point(&self, data: Data, point: &[Num; 3]) -> Data {
-        let &[x, y, z] = point;
-        let z = self.map_range(z);
-        data.move_to((x, y))
-            .move_by((0, -z))
-            .line_by((-z, z))
-            .line_by((z, z))
-            .line_by((z, -z))
-            .line_by((-z, -z))
+    /// This is a scriber that uses points to construct `Data` to build a `Path` and add it to a `SVG` document. It draws
+    /// diamonds around the supplied points
+    fn scribe_points(&self, doc: SVG, points: &Vec<[Num; 3]>, scriber: &Scriber) -> SVG {
+        let data = points.into_iter().fold(Data::new(), |d, &point| {
+            let [x, y, z] = point;
+            let z = self.map_range(z);
+            d.move_to((x, y))
+                .move_by((0, -z))
+                .line_by((-z, z))
+                .line_by((z, z))
+                .line_by((z, -z))
+                .line_by((-z, -z))
+        });
+
+        let path = Path::new()
+            .set("fill", scriber.fill)
+            .set("stroke", scriber.stroke)
+            .set("stroke-width", scriber.stroke_width)
+            .set("d", data);
+        doc.add(path)
     }
 }
 
-//This could later on use different strategies to visualize a point
-//ie. perfect, arc, something else interesting
+pub struct CircleScriber {}
+
+impl HoloPointStrategy for CircleScriber {
+    /// This is a scriber that uses `Element`s (namely `Circle`s) and adds those directly to the `SVG` document.
+    /// It draws circles around each point, scaled by the z value. (Larger z = farther away = larger circle with flatter arc)
+    fn scribe_points(&self, mut doc: SVG, points: &Vec<[Num; 3]>, scriber: &Scriber) -> SVG {
+        for point in points {
+            let &[x, y, z] = point;
+
+            // We scale the points by half the canvas size
+            let circle = Circle::new()
+                .set("cx", x * scriber.canvas_size.width as f32 * 0.5)
+                .set("cy", y * scriber.canvas_size.height as f32 * 0.5)
+                .set("r", z * scriber.canvas_size.width as f32 * 0.5)
+                .set("stroke-width", scriber.stroke_width)
+                .set("stroke", scriber.stroke)
+                .set("fill", scriber.fill);
+
+            doc = doc.add(circle);
+        }
+        doc
+    }
+}
+
+/// Different strategies to visualize a point
 pub trait HoloPointStrategy {
-    fn scribe_point(&self, data: Data, point: &[Num; 3]) -> Data;
+    fn scribe_points(&self, doc: SVG, points: &Vec<[Num; 3]>, scriber: &Scriber) -> SVG;
 }
 
 pub struct Scriber {
-    //Ammount to increase the document view in x/y beyond furthest away points
-    //This effects if some arcs are drawn outisde the work area
-    margin: Num,
     stroke: &'static str,
     stroke_width: Num,
+    fill: &'static str,
     point_scribing_strategy: Box<dyn HoloPointStrategy>,
+    canvas_size: CanvasSize,
+    margin: Num,
 }
 
 type Num = f32;
```

Comment:
I spent a long time wondering if there was a `Num` crate I was missing 😂 Maybe put this up at the top of the file so newbs like me can find it.

Also probably good practice to use this throughout, in case we ever get up a creek and need double precision floats